glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -193,6 +195,10 @@ open class GleanInternalAPI internal constructor () {
                 // iterating over and deleting the pings doesn't happen at the same time.
                 synchronized(PingUploader.pingQueueLock) {
                     LibGleanFFI.INSTANCE.glean_set_upload_enabled(handle, enabled.toByte())
+
+                    if (!enabled) {
```

Comment:
nit: please add a small comment mentioning why we care about cancelling workers here.